CDCgov/prime-simplereport
# Map syphilis history to FHIR

# BACKEND PULL REQUEST

## Related Issue

- Resolves #7564 

## Changes Proposed

- maps syphilis history to FHIR
- bulk upload results
  - adds validation to the `syphilis_history` column so that it is [one of these values](https://github.com/CDCgov/prime-simplereport/blob/923e291b3aeff3a67652f8c3c181cb42cff1bf0f/backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java#L185)

## Additional Information
- [see context here](https://skylight-hq.slack.com/archives/C0411VC78DN/p1727372370949959?thread_ts=1721667117.540449&cid=C0411VC78DN) about how the shape of the FHIR message for this observation 

## Testing
- Use this branch `elisa/with-log-8100-fhir-syphilis-aoe` to test uploading syphilis results or entering syphilis results individually
  - search in your backend console `ZIPZAP` to see the shape of the msg
